vcd/resource_vcd_vapp_vm.go

```diff
@@ -834,6 +883,15 @@ func genericResourceVmCreate(d *schema.ResourceData, meta interface{}, vmType ty
 			}
 		}
 
+		if enterBiosSetup, ok := d.Get("boot_options.0.enter_bios_setup").(bool); ok {
+			biosSetup := &types.BootOptions{
+				EnterBiosSetup: &enterBiosSetup}
+
+			_, err = vm.UpdateBootOptions(biosSetup)
+			if err != nil {
+				return diag.Errorf("error enabling BIOS setup: %s", err)
+			}
+		}
```

Comment:
This is made so that after the boot, we get to the requested state. Without this piece, if a user asks for `enter_bios_setup` to be true and changes some other option that requires a VM power cycle, it would result in `enter_bios_setup` to be set to `false`

Reply:
Thanks, I will attempt to do some manual checks here as well